app/stac_api/validators_serializer.py

```diff
@@ -521,3 +509,36 @@ def validate_intersects(self, geojson):
             message = f"Invalid query: " \
                 f"Could not transform {geojson} to a geometry; {error}"
             self.errors['intersects'] = _(message)
+
+    def validate_query_parameters_post_search(self, query_param):
+        '''Validates the query parameters for POST requests on the search endpoint.
+        If any invalid query parameters are found, the dict self.errors will be extended
+        with the corresponding message.
+
+        Args:
+            query_param: dict
+                Copy of the harmonized QueryDict
+        '''
+        accepted_query_parameters = [
+            "bbox", "collections", "datetime", "ids", "intersects", "limit", "query"
+        ]
+        # make sure that all queried parameters (list(query_param.keys())) are in
+        # the accepted_query_parameters
+        if not all(
+            parameter in accepted_query_parameters for parameter in list(query_param.keys())
```

Comment:
`... for parameter in query_param.keys():` and even `... for parameter in query_param:` is sufficient (no need for the `list`)